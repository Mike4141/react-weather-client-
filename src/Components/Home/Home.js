import React, { Component } from "react";
import Weather from "../Weather/Weather";

class Home extends Component {

  render() {

    return (
      <div>
        <div className="navbar">
          <link
            href="https://fonts.googleapis.com/css?family=Bangers&display=swap"
            rel="stylesheet"
          />
          <link
            href="https://fonts.googleapis.com/css?family=Bangers|Bowlby+One+SC&display=swap"
            rel="stylesheet"
          />
          <h2>My Weather</h2>
        </div>

        <main>
          <div className="bg">
            <form onSubmit={this.props.handleAddCity}>
              <div className="w-header">
                <h1>My Weather</h1>
              </div>

              <div className="changes">
                <p>The current weather for your favorite cities!</p>
              </div>

              <div className="changes">
                {this.props.invalidSearchError && <p>invalid search</p>}

              </div>

              <div className="changes">
                {this.props.duplicateError && <p>City Already Exist</p>}
              </div>

              <div className="flex">
                <input
                  placeholder="New city name..."
                  type="text"
                  value=""
                  value={this.props.newCityName}
                  onChange={this.props.handleInputChange}
                  required
                />

                <button className="btn" color="primary">
                  Add City
                </button>
              </div>
            </form>
          </div>
        </main>

        <div className="select">
          {this.props.cityList.length === 0 && <li>No cities added yet.</li>}
          {this.props.cityList.length > 0 && <li>Select a city.</li>}
        </div>

        <div className="cities">
          <ul>
            {this.props.cityList.map(
              (city, i) =>
                console.log(city) || (
                  <li className="ct" key={i}>
                    <span
                      className="selector"
                      onClick={() => this.props.handleChangeCity(city.name)}
                    >
                      {city.name}
                    </span>

                    <button
                      className="delete-btn"
                      onClick={() => this.props.handleDelete(city.id)}
                    >
                      Delete
                    </button>
                  </li>
                )
            )}
          </ul>
        </div>

        <Weather data={this.props.weather} />
      </div>
    );
  }
}

export default Home;
