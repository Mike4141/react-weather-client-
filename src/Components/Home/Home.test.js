import React from 'react';
import ReactDOM from 'react-dom';
import Home from './Home';
import renderer from 'react-test-renderer';

//test data
const props = {
    invalidSearchError: false,
    duplicateError: false,
    handleInputChange: () => { },
    handleAddCity: () => { },
    handleChangeCity: () => { },
    handleDelete: () => { },
    newCityName: '',
    cityList: [],
    weather: ''
}

const cityList = [
    {
        "id": 2,
        "name": "New York",
        "created_at": "2020-04-22T14:08:11.002Z"
    },
    {
        "id": 3,
        "name": "Oklahoma",
        "created_at": "2020-04-22T14:09:00.764Z"
    },
    {
        "id": 9,
        "name": "Toronto",
        "created_at": "2020-04-22T15:13:28.699Z"
    },
    {
        "id": 10,
        "name": "Seattle",
        "created_at": "2020-04-22T15:15:17.424Z"
    },
    {
        "id": 11,
        "name": "dallas",
        "created_at": "2020-04-22T16:15:20.228Z"
    }
]

// smoke tests
it('renders without crashing with inititial state', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Home {...props} />, div);
    // clean up code
    ReactDOM.unmountComponentAtNode(div);

})

it('renders without crashing with city list', () => {
    props.cityList = cityList
    const div = document.createElement('div');
    ReactDOM.render(<Home {...props} />, div);
    // clean up code
    ReactDOM.unmountComponentAtNode(div);

})

//snapshot tests
test('Home component with initial data', () => {
    const component = renderer.create(
        <Home {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});


test('Home component on duplicate entry', () => {
    props.duplicateError = true
    const component = renderer.create(
        <Home {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});


test('Home component on invalid search', () => {
    props.invalidSearchError = true
    const component = renderer.create(
        <Home {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

