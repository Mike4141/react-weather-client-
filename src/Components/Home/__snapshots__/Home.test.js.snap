// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component on duplicate entry 1`] = `
<div>
  <div
    className="navbar"
  >
    <link
      href="https://fonts.googleapis.com/css?family=Bangers&display=swap"
      rel="stylesheet"
    />
    <link
      href="https://fonts.googleapis.com/css?family=Bangers|Bowlby+One+SC&display=swap"
      rel="stylesheet"
    />
    <h2>
      My Weather
    </h2>
  </div>
  <main>
    <div
      className="bg"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="w-header"
        >
          <h1>
            My Weather
          </h1>
        </div>
        <div
          className="changes"
        >
          <p>
            The current weather for your favorite cities!
          </p>
        </div>
        <div
          className="changes"
        />
        <div
          className="changes"
        >
          <p>
            City Already Exist
          </p>
        </div>
        <div
          className="flex"
        >
          <input
            onChange={[Function]}
            placeholder="New city name..."
            required={true}
            type="text"
            value=""
          />
          <button
            className="btn"
            color="primary"
          >
            Add City
          </button>
        </div>
      </form>
    </div>
  </main>
  <div
    className="select"
  >
    <li>
      Select a city.
    </li>
  </div>
  <div
    className="cities"
  >
    <ul>
      <li
        className="ct"
      >
        <span
          className="selector"
          onClick={[Function]}
        >
          New York
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
      <li
        className="ct"
      >
        <span
          className="selector"
          onClick={[Function]}
        >
          Oklahoma
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
      <li
        className="ct"
      >
        <span
          className="selector"
          onClick={[Function]}
        >
          Toronto
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
      <li
        className="ct"
      >
        <span
          className="selector"
          onClick={[Function]}
        >
          Seattle
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
      <li
        className="ct"
      >
        <span
          className="selector"
          onClick={[Function]}
        >
          dallas
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
    </ul>
  </div>
  <div />
</div>
`;

exports[`Home component on invalid search 1`] = `
<div>
  <div
    className="navbar"
  >
    <link
      href="https://fonts.googleapis.com/css?family=Bangers&display=swap"
      rel="stylesheet"
    />
    <link
      href="https://fonts.googleapis.com/css?family=Bangers|Bowlby+One+SC&display=swap"
      rel="stylesheet"
    />
    <h2>
      My Weather
    </h2>
  </div>
  <main>
    <div
      className="bg"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="w-header"
        >
          <h1>
            My Weather
          </h1>
        </div>
        <div
          className="changes"
        >
          <p>
            The current weather for your favorite cities!
          </p>
        </div>
        <div
          className="changes"
        >
          <p>
            invalid search
          </p>
        </div>
        <div
          className="changes"
        >
          <p>
            City Already Exist
          </p>
        </div>
        <div
          className="flex"
        >
          <input
            onChange={[Function]}
            placeholder="New city name..."
            required={true}
            type="text"
            value=""
          />
          <button
            className="btn"
            color="primary"
          >
            Add City
          </button>
        </div>
      </form>
    </div>
  </main>
  <div
    className="select"
  >
    <li>
      Select a city.
    </li>
  </div>
  <div
    className="cities"
  >
    <ul>
      <li
        className="ct"
      >
        <span
          className="selector"
          onClick={[Function]}
        >
          New York
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
      <li
        className="ct"
      >
        <span
          className="selector"
          onClick={[Function]}
        >
          Oklahoma
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
      <li
        className="ct"
      >
        <span
          className="selector"
          onClick={[Function]}
        >
          Toronto
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
      <li
        className="ct"
      >
        <span
          className="selector"
          onClick={[Function]}
        >
          Seattle
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
      <li
        className="ct"
      >
        <span
          className="selector"
          onClick={[Function]}
        >
          dallas
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
    </ul>
  </div>
  <div />
</div>
`;

exports[`Home component with initial data 1`] = `
<div>
  <div
    className="navbar"
  >
    <link
      href="https://fonts.googleapis.com/css?family=Bangers&display=swap"
      rel="stylesheet"
    />
    <link
      href="https://fonts.googleapis.com/css?family=Bangers|Bowlby+One+SC&display=swap"
      rel="stylesheet"
    />
    <h2>
      My Weather
    </h2>
  </div>
  <main>
    <div
      className="bg"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="w-header"
        >
          <h1>
            My Weather
          </h1>
        </div>
        <div
          className="changes"
        >
          <p>
            The current weather for your favorite cities!
          </p>
        </div>
        <div
          className="changes"
        />
        <div
          className="changes"
        />
        <div
          className="flex"
        >
          <input
            onChange={[Function]}
            placeholder="New city name..."
            required={true}
            type="text"
            value=""
          />
          <button
            className="btn"
            color="primary"
          >
            Add City
          </button>
        </div>
      </form>
    </div>
  </main>
  <div
    className="select"
  >
    <li>
      Select a city.
    </li>
  </div>
  <div
    className="cities"
  >
    <ul>
      <li
        className="ct"
      >
        <span
          className="selector"
          onClick={[Function]}
        >
          New York
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
      <li
        className="ct"
      >
        <span
          className="selector"
          onClick={[Function]}
        >
          Oklahoma
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
      <li
        className="ct"
      >
        <span
          className="selector"
          onClick={[Function]}
        >
          Toronto
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
      <li
        className="ct"
      >
        <span
          className="selector"
          onClick={[Function]}
        >
          Seattle
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
      <li
        className="ct"
      >
        <span
          className="selector"
          onClick={[Function]}
        >
          dallas
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
    </ul>
  </div>
  <div />
</div>
`;
